Document how to access the Hive's installation logs (#165)
openshift/ci-docs: content/en/docs/how-tos/cluster-claim.md

  name: dptp-demo-cluster-pools
```

### Set up RBACs for debugging

The `ClusterRole` `hive-cluster-pool-admin` will [contains the permissions](https://github.com/openshift/hive/blob/master/docs/clusterpools.md#managing-admins-for-cluster-pools) to all resources created for a pool.

```yaml
apiVersion: user.openshift.io/v1
kind: Group
metadata:
  name: dptp-demo-cluster-pools-admins
users:
- petr
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: dptp-demo-cluster-pool-admin
  namespace: dptp-demo-cluster-pools
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: hive-cluster-pool-admin
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: dptp-demo-cluster-pools-admins
---

> # the pool owners need the permission of listing the namespace to filter out which namespaces are created for their pools

Reviewer: ```suggestion
# the pool owners need the following cluster permissions to select namespaces created for their pools
```

type: Opaque
```

> ### Set up RBACs for debugging

Reviewer 2: Let's add this as a second step, after creating the namespace, so that once-per-namespace actions are together.

The `ClusterRole` `hive-cluster-pool-admin` will [contains the permissions](https://github.com/openshift/hive/blob/master/docs/clusterpools.md#managing-admins-for-cluster-pools) to all resources created for a pool.

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: hive-cluster-pool-admin-0
  namespace: dptp-demo-cluster-pools
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: hive-cluster-pool-admin
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: petr
```

The pool owners need the permission of listing the namespace to filter out which namespaces are created for their pools.
Login to [the `hive` cluster](https://console-openshift-console.apps.hive.9xw5.p1.openshiftapps.com/dashboards) and add the user into

> [the cluster-namespace-viewer group](https://github.com/openshift/release/blob/master/clusters/hive/admin_cluster-namespace-view_rbac.yaml).

Reviewer 2: This is confusing to me. When we instruct users to create one `RoleBinding` already, could we make the process entirely straighforward by just instrcut them to create a 1) Group 2) RB that group to `hive-cluster-pool-admin` 3) CRB that group to `cluster-namespace-view` here, in a single step?

I also don't understand why we instruct the users to "login to the cluster and add the user into the group": the login seems to be a dead end, adding the user is GitOps.

Author: I will create another PR to follow the instructions for hypershift.

Removed the login instruction.
I was trying to tell how to determine the username by login.

Author: https://github.com/openshift/release/pull/19855

The cluster pools are maintained by Hive behind the scenes, so installation failures, cloud platform account
misconfigurations and similar issues are not exposed to actual CI jobs: the jobs will simply never successfully claim a
cluster if Hive fails to install them.

> The `ClusterRole` `hive-cluster-pool-admin` will [contains the permissions](https://github.com/openshift/hive/blob/master/docs/clusterpools.md#managing-admins-for-cluster-pools) to all resources created for a pool.

Reviewer 2: I'd split this in two parts:

1) Add a section to the "add a new pool" procedure where you instruct new pool conributors to add the RBAC manifest, with a very brief eplanation that this will allow cluster pool owners to troubleshoot/inspect how Hive installs clusters in their pool.
2) Add a paragraph here that describes how to actually troubleshoot a pool, provided you are subject to the RBAC added in 1)

Author: Updated.

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: hive-cluster-pool-admin-0
  namespace: dptp-demo-cluster-pools
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: hive-cluster-pool-admin
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: petr
```

The pool owners can list the namespace to filter out which namespaces are created for their pools.
Login to [the `hive` cluster](https://console-openshift-console.apps.hive.9xw5.p1.openshiftapps.com/dashboards) and add the user into

> [the cluster-namespace-viewer group](https://github.com/openshift/release/tree/master/clusters/hive/pools/hypershift/hypershift-pool_admins.yaml).

Reviewer 2: For the actual troubleshooting, let's assume that the user is already subject to the RBAC; if not, they need to be added first via gitops. Let's not link to an actual hypershift RBACs.

Author: I linked to the wrong file.
Will update.

The installation logs can be found in the `hive` container logs in the namespaces returned by the following command.

```console

> $ oc get ns | grep ${cluster_pool_name}

Reviewer 2: So this gets you namespaces, can you add the commands that shows the pods inside and then the correct `oc logs` for the given pod?

Author: Updated.

## Troubleshooting Cluster Pools

> ### Access Cluster Installation logs

Reviewer: ```suggestion
### Accessing Cluster Installation logs
```

The cluster pools are maintained by Hive behind the scenes, so installation failures, cloud platform account
misconfigurations and similar issues are not exposed to actual CI jobs: the jobs will simply never successfully claim a
cluster if Hive fails to install them.

The installation logs can be found in the `hive` container logs with the following commands:

```console
# filter out the namespaces start with the cluster pool's name

> $ oc get ns | grep dptp-demo-cluster-pool

Reviewer: Do these not get annotations or labels or anything to do something more intelligent for filtering here?

Author: Perfect! It has the label.

```console
oc --context hive get ns ci-ocp-4-8-amd64-aws-us-east-1-b94kw -o yaml
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    openshift.io/sa.scc.mcs: s0:c39,c9
    openshift.io/sa.scc.supplemental-groups: 1001500000/10000
    openshift.io/sa.scc.uid-range: 1001500000/10000
  creationTimestamp: "2021-07-01T19:09:09Z"
  labels:
    hive.openshift.io/cluster-pool-name: ci-ocp-4-8-amd64-aws-us-east-1
    kubernetes.io/metadata.name: ci-ocp-4-8-amd64-aws-us-east-1-b94kw
```

Author: I will update tomorrow.

Author: Done.

dptp-demo-cluster-pool-pclkt              Active   16m

# get the provision pod name
$ oc get pod -n dptp-demo-cluster-pool-pclkt
NAME                                                            READY   STATUS     RESTARTS   AGE
dptp-demo-cluster-pool-pclkt-0-h5tml-provision-m2pp9   1/3     NotReady   0          18m

# access the logs

> $ oc logs -n dptp-demo-cluster-pool-pclkt dptp-demo-cluster-pool-pclkt-0-h5tml-provision-m2pp9 -c hive

Reviewer: Similarly... I would either present a script here or a one-liner with expansions `$()` - not a set of commands where copy-paste errors occur.

Author: See if it looks better now.

## Troubleshooting Cluster Pools

### Accessing Cluster Installation logs
The cluster pools are maintained by Hive behind the scenes, so installation failures, cloud platform account
misconfigurations and similar issues are not exposed to actual CI jobs: the jobs will simply never successfully claim a
cluster if Hive fails to install them.

The installation logs can be found in the `hive` container logs with the following commands:

```console
# filter out the namespaces start with the cluster pool's name

> $ oc get ns -l hive.openshift.io/cluster-pool-name=dptp-demo-cluster-pool

Author: Updated.

dptp-demo-cluster-pool-pclkt              Active   16m

# get the provision pod name
$ oc get pod -n dptp-demo-cluster-pool-pclkt
NAME                                                            READY   STATUS     RESTARTS   AGE
dptp-demo-cluster-pool-pclkt-0-h5tml-provision-m2pp9   1/3     NotReady   0          18m
...

# select a namespace
$ ns=dptp-demo-cluster-pool-pclkt-0-h5tml-provision-m2pp9
# access the logs

> $ oc -n $ns logs $(oc -n $ns get pod --sort-by=.metadata.creationTimestamp -l hive.openshift.io/cluster-deployment-name=$ns --no-headers -o custom-columns=":metadata.name" | head -n 1) -c hive

Reviewer: ```
$ namespace="$( oc get namespaces --selector hive.openshift.io/cluster-pool-name=dptp-demo-cluster-pool -o jsonpath={.metadata.name} | tr ' ' $'\n' | head -n 1 )"
$ pod="$( oc --namespace "${namespace}" get pods --selector hive.openshift.io/cluster-deployment-name="${namespace}" -o jsonpath={.items[*].metadata.name} | tr ' ' $'\n' | head -n 1 )"
$ oc --namespace  "${namespace}" logs "${pod}" -c hive
```

Author: Updated.
